src/content/blog/2025-09-05-memorable-wallet-addresses-custom-domain.md

```diff
@@ -0,0 +1,248 @@
+
+### `MonetizationEvent`
+
+Let's say, instead of using a wallet address that our provider gave us, we alias it to a custom address as shown above, and add it to our webpage. This sort of indirection can also arise from using the [probabilistic revenue sharing generator](https://webmonetization.org/tools/prob-revshare/), or from your wallet provider may be using an aliased wallet address itself.
+
+When a `monetization` event is emitted, how do we know what wallet address was actually used? And how do we know what wallet address we originally provided? This is even more relevant if your webpage includes multiple monetization link tags.
+
+Thankfully, the `MonetizationEvent` includes both these details!
+
+Let's look at the [`MonetizationEvent`](https://webmonetization.org/specification/#dom-monetizationevent) in detail.
+
+```webidl
+interface MonetizationEvent : Event {
+  readonly attribute MonetizationCurrencyAmount amountSent;
+  readonly attribute USVString paymentPointer;
+  readonly attribute USVString? incomingPayment;
+  // other details not relevant in this context
+}
+```
+
+Here, `paymentPointer` is the resolved wallet address that was used. For example, with a custom domain payment pointer - like I've mapped `$sidvishnoi.com` to resolve to my GateHub wallet address (`https://ilp.gatehub.net/981946513/eur`) above - the `paymentPointer` will resolve to my GateHub address. When using a wallet address from the probabilistic revenue sharing tool, it'll correspond to the wallet address that was chosen randomly.
+
+And how do we get the original wallet address?
+
+If you look closely, the `MonetizationEvent` inherits the `Event` interface. So, everything that belongs to `Event` is also part of `MonetizationEvent`. The part we're looking for here is the event's `target` attribute. This target corresponds to the `<link rel="monetization">` element where you added your wallet address. The link element has the `href` attribute, which corresponds to the original wallet address we provided on our page. We can get the original wallet address as follows:
+
+```html
+<html>
+  <head>
+    <link rel="monetization" href="https://sidvishnoi.com/.well-known/pay" />
+  </head>
+</html>
+```
+
+```ts
+window.addEventListener("monetization", (event) => {
+  const linkElement: HTMLLinkElement = event.target;
+  const originalWalletAddress = linkElement.href;
+  // -> https://sidvishnoi.com/.well-known/pay
+
+  const usedWalletAddress = event.paymentPointer;
+  // -> https://ilp.gatehub.net/981946513/eur
+});
+```
+
+Depending on the use case, you may care about either or both of the wallet addresses. I hope it was helpful for you to know how to get each.
+
+## Closing words
+
+I look forward to seeing wallet addresses on your own domains! Remember, `$sidvishnoi.com` is my payment pointer 😉
```

Comment:
I'd end off with your wallet address instead of your payment pointer. Small difference.